docs/dev/migration.md

```diff
@@ -25,6 +25,62 @@ under the License.
 * This will be replaced by the TOC
 {:toc}
 
+## Migrating from Flink 1.2 to Flink 1.3
+
+There are a few APIs that have been changed since Flink 1.2. Most of the changes are documented in their
+specific documentations. The following is a consolidated list of API changes and links to details for migration when
+upgrading to Flink 1.3.
+
+### `TypeSerializer` interface changes
+
+This would be relevant mostly for users implementing custom `TypeSerializer`s for their state.
+
+Since Flink 1.3, two additional methods have been added that are related to serializer compatibility
+across savepoint restores. Please see
+[Handling serializer upgrades and compatibility](../stream/state.html#handling-serializer-upgrades-and-compatibility)
+for further details on how to implement these methods.
+
+### `ProcessFunction` is always a `RichFunction`
+
+In Flink 1.2, `ProcessFunction` and its rich variant `RichProcessFunction` was introduced.
+Since Flink 1.3, `RichProcessFunction` was removed and `ProcessFunction` is now always a `RichFunction` with access to
+the lifecycle methods and runtime context.
+
+### Flink CEP library API changes
+
+The CEP library in Flink 1.3 ships with a number of new features which have led to some changes in the API.
+Please visit the [CEP Migration docs](../libs/cep.html#migrating-from-an-older-flink-version) for details.
+
+### Table API Changes
+
+*TBA*
+
+### Queryable State client construction changes
+
+*TBA*
```

Comment:
I don't think these TBA sections are a good idea in the documentation.

Maybe we should comment them out before committing them, so that they don't show up in the page